src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11448,160 +11457,126 @@ bool isBinaryPatternKeyword()
             }
         }
 
+        /// <summary>
+        /// Tokens that match the following are considered a possible lambda expression:
+        /// <code>attribute-list* ('async' | 'static')* type? ('(' | identifier) ...</code>
+        /// For better error recovery 'static =>' is also considered a possible lambda expression.
+        /// </summary>
```

Comment:
we need to state (and ideally check somehow) that we *must* be in a resettable location.  that's because this code actually consumes the input and throws it away, and thus would corrupt any caller who was actually building thr tree.  it's fine to do this if we're in a reset block though as we'll then jump back and resume parsing when done.

Reply:
> we need to state (and ideally check somehow) that we must be in a resettable location

I'm not sure I understand what it means to be in a resettable location. But `IsPossibleLambdaExpression()` and `IsPossibleLambdaExpressionCore()` previously used a `ResetPoint` in some code paths. The change is really to use a `ResetPoint` in all code paths.

Reply:
ignore what i said.  something about the diff confused me and i thought the reset point was moved *out* of this method into the caller.  As it's in this method itself, this seems totally fine to me :)